### PR TITLE
Feature/zcs 2976

### PR DESCRIPTION
**Pull request by grishick**
Implemented "test" method that fetches some contacts and parses them. Added a SOAP test for the 'test' method.

### CODE REVIEW

File: src/java/com/synacor/zimbra/ys/contacts/YahooContactsImport.java [hunk 1]
```diff
@@ -45,39 +47,52 @@ public void test() throws ServiceException {
...
+                            if(contact != null) {
+                                testContact = contact;
+                                break;
+                            }
                         }
+                    } else {
+                        ZimbraLog.extensions.debug("Did not find 'contact' element in 'contacts' object");
                     }
+                } else {
+                    ZimbraLog.extensions.debug("Did not find 'contats' element in JSON response object");
```
Comment (fatkudu) [inline src/java/com/synacor/zimbra/ys/contacts/YahooContactsImport.java]: typo - contats

File: src/java/com/synacor/zimbra/ys/contacts/YahooContactsImport.java [hunk 2]
```diff
@@ -45,39 +47,52 @@ public void test() throws ServiceException {
...
-        if(jsonContacts == null || jsonContacts.size() == 0) {
+        if(testContact == null) {
             int respCode = 0;
             String respLine = "";
             if(response != null) {
                 respCode = response.getStatusLine().getStatusCode();
                 respLine = response.getStatusLine().getReasonPhrase();
             }
-            throw ServiceException.FAILURE(String.format("Failed to fetch contacts from  Yahoo Contacts API for testing. Response status code %d. Respose status line: %s", respCode, respLine), null);
+            throw ServiceException.FAILURE(String.format("Data source test failed. Failed to fetch contacts from  Yahoo Contacts API for testing. Response status code %d. Respose status line: %s. Response body %s", respCode, respLine, respContent), null);
```
Comment (fatkudu) [inline src/java/com/synacor/zimbra/ys/contacts/YahooContactsImport.java]: typo Respose - although I guess this is temporary code?

**Review by fatkudu — APPROVED** [review]
Trivial suggested changes.

File: src/java/com/synacor/zimbra/ys/contacts/YahooContactsImport.java (hunk 1 above)
Comment (tmclane) [reply src/java/com/synacor/zimbra/ys/contacts/YahooContactsImport.java]: 👍

**Review by tmclane — APPROVED** [review]

**Pull request merged**